BrowserKit/Sources/WebEngine/WKWebview/WKEngineSession.swift

```diff
@@ -55,23 +56,40 @@ class WKEngineSession: EngineSession {
             webView.loadFileURL(url, allowingReadAccessTo: url.deletingLastPathComponent())
             return
         }
+
         webView.load(request)
+        logger.log("Loaded request", level: .debug, category: .webview)
     }
 
     func stopLoading() {
         webView.stopLoading()
+        logger.log("Stop loading", level: .debug, category: .webview)
     }
 
     func reload() {
-        // TODO: FXIOS-7906 #17650 Handle reload in WKEngineSession
+        // If the current page is an error page load the original URL
+        if let url = webView.url,
+            let internalUrl = WKInternalURL(url),
+            let page = internalUrl.originalURLFromErrorPage {
```

Comment:
nit: alignment

Reply:
🤦‍♀️ I'll fix it in my next PR as the build is green. Thank you!